### PR TITLE
Connect to backend

### PR DESCRIPTION
**Pull request by HishamAlmannai**

### CODE REVIEW

File: pages/api/archive.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export default async function handler(req, res) {
+	if (req.method === 'POST') {
+		const oldTaskData = JSON.parse(req.body);
+		await dbConnect();
+		await Archive.insertMany(oldTaskData);
+		const deleteMany = await Task.deleteMany({ done: true });
+
+		console.log(req.body);
+		res.status(200).json({
```
Comment (ThomasF85) [inline pages/api/archive.js:13]: A successful post request should return a status code 201 (CREATED)

File: pages/api/archive.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { dbConnect } from '../../src/lib/db/database';
+import Archive from '../../src/lib/models/Archive';
+import Task from '../../src/lib/models/Task';
+
+export default async function handler(req, res) {
+	if (req.method === 'POST') {
+		const oldTaskData = JSON.parse(req.body);
+		await dbConnect();
+		await Archive.insertMany(oldTaskData);
+		const deleteMany = await Task.deleteMany({ done: true });
```
Comment (ThomasF85) [inline pages/api/archive.js:10]: It seems like you really update existing data here. A post request should only be used when creating a new resource.

File: pages/api/task/[id].js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export default async function handler(req, res) {
+	const { id } = req.query;
+
+	if (req.method === 'DELETE') {
+		const deletedTask = await Task.findByIdAndDelete(id);
+		res.status(200).json({
+			message: 'task deleted',
+			entry: deletedTask,
+		});
+		[];
```
Comment (ThomasF85) [inline pages/api/task/[id].js:13]: useless expression. please delete

File: pages/api/task/[id].js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		const newTaskData = JSON.parse(req.body);
+		await dbConnect();
+		const updatedTask = await Task.findByIdAndUpdate(id, newTaskData, {
+			new: true,
+		});
+		res.status(200).json({
+			message: 'task edited',
+			entry: updatedTask,
+		});
+	} else {
```
Comment (ThomasF85) [inline pages/api/task/[id].js:24]: PATCH and POST requests will run into this else block. They should not.

File: pages/api/task/create.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+		await dbConnect();
+
+		const newTask = await Task.create({
+			name: newTaskData.name,
+			done: newTaskData.done,
+			dueDate: newTaskData.dueDate,
+			doneDate: newTaskData.doneDate,
+			startDate: newTaskData.startDate,
+		});
+		console.log(req.body);
```
Comment (ThomasF85) [inline pages/api/task/create.js:16]: Please delete console.log statements that are used for debugging during development.

**Review by ThomasF85 — CHANGES_REQUESTED** [review]

File: pages/api/task/[id].js (hunk 3 above)
Comment (HishamAlmannai) [reply pages/api/task/[id].js:13]: done. thx

File: pages/api/archive.js (hunk 1 above)
Comment (HishamAlmannai) [reply pages/api/archive.js:13]: True & done.
thx

File: pages/api/archive.js (hunk 2 above)
Comment (HishamAlmannai) [reply pages/api/archive.js:10]: Changed to PUT. Thanks

File: pages/api/task/create.js (hunk 5 above)
Comment (HishamAlmannai) [reply pages/api/task/create.js:16]: I thought these where for Serverside visualization. Thanks again

File: pages/api/task/[id].js (hunk 4 above)
Comment (HishamAlmannai) [reply pages/api/task/[id].js:24]: Removed it. These are very nice details. Thanks

**Comment by HishamAlmannai** [comment]
LGTM

**Comment by HishamAlmannai** [comment]
Finally?

**Pull request merged**